### PR TITLE
cephadm/mgr: adding logic to handle --no-overwrite for tuned profiles

### PR DESCRIPTION
**Pull request by rkachach**
Fixes: https://tracker.ceph.com/issues/57032

Signed-off-by: Redouane Kachach <rkachach@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by rkachach** [comment]
jenkins retest this please

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -2536,11 +2536,14 @@ def _apply(self, spec: GenericSpec) -> str:
...
 
     @handle_orch_error
-    def apply_tuned_profiles(self, specs: List[TunedProfileSpec]) -> str:
+    def apply_tuned_profiles(self, specs: List[TunedProfileSpec], no_overwrite: bool = False) -> str:
         outs = []
         for spec in specs:
-            self.tuned_profiles.add_profile(spec)
-            outs.append(f'Saved tuned profile {spec.profile_name}')
+            if no_overwrite and self.tuned_profiles.exists(spec.profile_name):
+                outs.append(f"Tuned profile '{spec.profile_name}' already exists (--no-overwrite is enabled)")
```
Comment (adk3798) [inline src/pybind/mgr/cephadm/module.py:2543]:
```suggestion
                outs.append(f"Tuned profile '{spec.profile_name}' already exists (--no-overwrite flag was passed")
```
NIT, I think it's a bit clearer

**Review by adk3798 — APPROVED** [review]

**Comment by adk3798** [comment]
jenkins test make check

**Comment by adk3798** [comment]
jenkins test api

**Comment by adk3798** [comment]
jenkins test dashboard cephadm

**Comment by adk3798** [comment]
we probably ought to document this functionality now that it's there, right @rkachach? It could be useful to people as the flag sort of make the command idempotent

**Comment by rkachach** [comment]
> we probably ought to document this functionality now that it's there, right @rkachach? It could be useful to people as the flag sort of make the command idempotent

Yep, it's a good idea to reflect it in the docs as well :+1:

File: doc/cephadm/host-management.rst [hunk 2]
```diff
@@ -281,6 +281,11 @@ given placement and `sysctl --system` will be run on the host.
   These settings are applied only at the host level, and are not specific
   to any certain daemon or container
 
+.. note::
+
+  Applying tuned profiles is idempotent when ``--no-overwrite`` option is passed.
```
Comment (anthonyeleven) [inline doc/cephadm/host-management.rst:286]: When the

**Review by anthonyeleven — APPROVED** [review]

**Review by anthonyeleven — CHANGES_REQUESTED** [review]
Change requested in comment

File: doc/cephadm/host-management.rst (hunk 2 above)
Comment (rkachach) [reply doc/cephadm/host-management.rst]: fixed

**Review by anthonyeleven — APPROVED** [review]

**Comment by rkachach** [comment]
jenkins test make check

**Comment by rkachach** [comment]
jenkins retest this please

**Review by adk3798 — APPROVED** [review]

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2022-09-02_11:54:06-orch:cephadm-wip-adk-testing-2022-09-01-2029-distro-default-smithi/

Failures tracked by

- https://tracker.ceph.com/issues/57371
- https://tracker.ceph.com/issues/55986
- https://tracker.ceph.com/issues/57287
- https://tracker.ceph.com/issues/57374

All failures are also present on main branch and not caused by PRs in the run

**Pull request merged**